From 7f29fa93fa3fe465b8afcd52fd662c45a68313b8 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Mon, 19 Nov 2012 14:53:57 -0800
Subject: [PATCH 228/628] spi: xilinx_qspips: Fix potential memory leaks

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit a8a16df3dbc8b208fcc68f69019bf736e44e502e

The spi_master struct was dynamically allocated but never freed.
Memory allocation used the type instead of *identifier to determine
the size of the area to allocate.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Acked-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/spi/spi-xilinx-qps.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/spi/spi-xilinx-qps.c b/drivers/spi/spi-xilinx-qps.c
index 0bcfc68..3e2b47b 100644
--- a/drivers/spi/spi-xilinx-qps.c
+++ b/drivers/spi/spi-xilinx-qps.c
@@ -899,7 +899,7 @@ static int __devinit xqspips_probe(struct platform_device *dev)
 	struct resource *r;
 	const unsigned int *prop;
 
-	master = spi_alloc_master(&dev->dev, sizeof(struct xqspips));
+	master = spi_alloc_master(&dev->dev, sizeof(*xqspi));
 	if (master == NULL)
 		return -ENOMEM;
 
@@ -1030,6 +1030,7 @@ release_mem:
 put_master:
 	platform_set_drvdata(dev, NULL);
 	spi_master_put(master);
+	kfree(master);
 	return ret;
 }
 
@@ -1069,6 +1070,7 @@ static int __devexit xqspips_remove(struct platform_device *dev)
 
 	spi_unregister_master(master);
 	spi_master_put(master);
+	kfree(master);
 
 	/* Prevent double remove */
 	platform_set_drvdata(dev, NULL);
-- 
1.7.5.4

